packages/@okta/vuepress-site/docs/guides/terraform-organize-configuration/main/index.md

```diff
@@ -0,0 +1,290 @@
+
+* Identify sensitive fields in variable definitions with the `sensitive` keyword so that it doesn't appear in the Terraform output:
+
+```hcl
+variable "private_key" {
+  description = "Client authentication information"
+  sensitive = true
+}
+```
+
+  > **Note:** Variables that are tagged with sensitive can't be used in loops. For more information on the `for_each` command, see [Reduce code duplication with loops](/docs/guides/terraform-syntax-tips/main/#reduce-code-duplication-with-loops).
+
+* Don't put secret values in a git repository or other source code control system. Anyone with access to a clone of the repository can access the secrets. To prevent saving secrets to the repository, add files that contain secrets to the git ignore file (`.gitignore`).
+
+* Don't store secrets in plain text in variable values files.
+
+    * Use key vault solutions that integrate with Terraform, such as the key vaults for each cloud provider or other key management products.
+
+    * Define secrets in system environment variables and reference them from Terraform. Set a system environment variable in a setup script with the prefix `TF_VAR_`. For related information, see [the Terraform article on environment variables](https://developer.hashicorp.com/terraform/cli/config/environment-variables).
+
+For example, for a variable `private_key`:
+
+```sh
+export TF_VAR_private_key=My_Example_Key_Value
+```
+
+Define a variable with an empty value, then reference it from the environment variable using the standard syntax: the `var` keyword followed by a period, followed by the variable name:
+
+```hcl
+variable "private_key" {}
+provider "okta" {
+  org_name = var.okta_org_name
+  base_url = var.base_url
+  client_id = var.client_id
+  scopes = var.scopes
+  private_key = var.private_key
+}
+```
+
+* Don't store Terraform state files locally, which is the default. These generated files can leak your credentials. Instead, use remote cloud storage. See the [Terraform documentation about remote backends](https://developer.hashicorp.com/terraform/language/backend/remote).
+
+For more strategies for secret storage in Terraform, see this [third-party article](https://blog.gruntwork.io/a-comprehensive-guide-to-managing-secrets-in-your-terraform-code-1d586955ace1).
+
+On a related note, several ways of performing authentication with Okta from Terraform and other SDKs exist. Okta recommends using key pairs. See [Enable Terraform access for your Okta org](/docs/guides/terraform-enable-org-access). Although Okta provides a feature called API keys, which work for authentication from external tools, Okta strongly recommends using public/private key pairs instead. Store the private key securely with one of the mechanisms described in this article to reduce the risk of unauthorized access. For related discussion, see [Why You Should Migrate to OAuth 2.0 From Static API Tokens](https://developer.okta.com/blog/2023/09/25/oauth-api-tokens).
```

Comment:
```suggestion
There are several ways to authenticate with Okta from Terraform and other SDKs. Okta recommends using key pairs. See [Enable Terraform access for your Okta org](/docs/guides/terraform-enable-org-access). Okta also recommends using public/private key pairs instead of API keys, which work for authentication from external tools. Store the private key securely with one of the mechanisms described in this article to reduce the risk of unauthorized access. For related discussion, see [Why You Should Migrate to OAuth 2.0 From Static API Tokens](https://developer.okta.com/blog/2023/09/25/oauth-api-tokens).
```